### PR TITLE
fix: fixed mock file system exception when using root path as a parameter

### LINKED ISSUE #702: MockFileSystem does not support Root directories
**Describe the bug**
Including a Root drive/directory in a MockFileSystem causes ArgumentNullException at line 82 of MockFileSystem.cs

**To Reproduce**
To recreate, execute the following line of code:

```csharp
var fileSystem = new MockFileSystem(new Dictionary<string, MockFileData>
{
    { @"c:\demo", new MockDirectoryData() },
    { @"z:\", new MockDirectoryData() },
    { @"Z:\foo", new MockDirectoryData() }
});
```

**Expected behavior**

"Z:\" should be a valid directory.

**Additional context**

```csharp
var fileSystem = new MockFileSystem(new Dictionary<string, MockFileData>
{
    { @"c:\demo", new MockDirectoryData() },
    //{ @"z:\", new MockDirectoryData() },
    { @"Z:\foo", new MockDirectoryData() }
});  //This produces no exception

var fileSystem = new MockFileSystem(new Dictionary<string, MockFileData>
{
    { @"c:\demo", new MockDirectoryData() },
    //{ @"z:\", new MockDirectoryData() },
    { @"Z:\foo", new MockDirectoryData() }
});  //This produces an exception.
```

### PR DESCRIPTION
**Pull request by vladimir-shirmanov**
# fix: MockFileSystem does not support Root directories

## Description

Added a check inside AddFile to be able to add new Drives within a constructor

Fixed #702

### CODE REVIEW

File: src/TestableIO.System.IO.Abstractions.TestingHelpers/MockFileSystem.cs [hunk 1]
```diff
@@ -243,6 +243,12 @@ public void AddFile(string path, MockFileData mockFile)
             }
 
             var directoryPath = Path.GetDirectoryName(fixedPath);
+            if (directoryPath == null)
```
Comment (vbreuss) [inline src/TestableIO.System.IO.Abstractions.TestingHelpers/MockFileSystem.cs:246]: I think, it is easier to understand, if you combine the two if-conditions, e.g.:
```csharp
            if (directoryPath == null)
            {
                AddDrive(fixedPath, new MockDriveData());
            }
            else if (!DirectoryExistsWithoutFixingPath(directoryPath))
            {
                // ...
            }
```

File: tests/TestableIO.System.IO.Abstractions.TestingHelpers.Tests/MockFileSystemTests.cs [hunk 2]
```diff
@@ -435,6 +435,26 @@ public void MockFileSystem_Constructor_ThrowsForNonRootedCurrentDirectory()
...
             Assert.That(ae.ParamName, Is.EqualTo("currentDirectory"));
         }
+        
+        [Test]
+        public void MockFileSystem_Constructor_ShouldSupportDifferentRootDrives()
+        {
+            var fileSystem = new MockFileSystem(new Dictionary<string, MockFileData>
+            {
+                [@"c:\"] = new MockDirectoryData(),
+                [@"z:\"] = new MockDirectoryData(),
```
Comment (vbreuss) [inline tests/TestableIO.System.IO.Abstractions.TestingHelpers.Tests/MockFileSystemTests.cs:445]: What happens, when you add a nested directory in a new drive, e.g. `z:\foo\bar`?
Could you adapt the test, to also verify that the drive is created in this case?

**Review by vbreuss — COMMENTED** [review]
Thanks for the pull request, I only have some minor suggestions.

Please also adapt the title to conform to the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) guideline.

**Comment by vladimir-shirmanov** [comment]
> Thanks for the pull request, I only have some minor suggestions.
> 
> Please also adapt the title to conform to the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) guideline.

oh sure, I'll add those in future commits. 
Haven't seen that instruction in CONTRIBUTION.md, sorry for that.

File: src/TestableIO.System.IO.Abstractions.TestingHelpers/MockFileSystem.cs (hunk 1 above)
Comment (vladimir-shirmanov) [reply src/TestableIO.System.IO.Abstractions.TestingHelpers/MockFileSystem.cs:246]: sure, fixed it

File: tests/TestableIO.System.IO.Abstractions.TestingHelpers.Tests/MockFileSystemTests.cs (hunk 2 above)
Comment (vladimir-shirmanov) [reply tests/TestableIO.System.IO.Abstractions.TestingHelpers.Tests/MockFileSystemTests.cs:445]: added more tests to cover those scenarios

File: src/TestableIO.System.IO.Abstractions.TestingHelpers/MockFileSystem.cs [hunk 3]
```diff
@@ -243,8 +243,12 @@ public void AddFile(string path, MockFileData mockFile)
             }
 
             var directoryPath = Path.GetDirectoryName(fixedPath);
-
-            if (!DirectoryExistsWithoutFixingPath(directoryPath))
+            if (directoryPath == null)
+            {
+                AddDrive(fixedPath, new MockDriveData());
+                SetEntry(fixedPath, mockFile);
```
Comment (vbreuss) [inline src/TestableIO.System.IO.Abstractions.TestingHelpers/MockFileSystem.cs:249]: Now you set the entry twice.
This line should be removed.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -12,3 +12,7 @@ The base version represents the MAJOR and MINOR parts of [SemVer](https://semver
 
 - `main` contains the latest sources. Each merge there triggers a deploy to `nuget.org`.
 - All versions on `nuget.org` have a matching GitHub release/tag.
+
```
Comment (vbreuss) [inline CONTRIBUTING.md:15]: Please don't update CONTRIBUTING.md as part of this pull request.
I am happy to accept a separate PR with the information.

Please note, that the requirement for Conventional Commits affects the pull request title and not the description (as the releases are automatically created based on this convention)!

**Review by vbreuss — CHANGES_REQUESTED** [review]

File: src/TestableIO.System.IO.Abstractions.TestingHelpers/MockFileSystem.cs (hunk 3 above)
Comment (vladimir-shirmanov) [reply src/TestableIO.System.IO.Abstractions.TestingHelpers/MockFileSystem.cs]: oh my bad, fixed

File: CONTRIBUTING.md (hunk 4 above)
Comment (vladimir-shirmanov) [reply CONTRIBUTING.md]: sure, I raised another PR with the updated contributing.md file here #1133

**Review by vbreuss — APPROVED** [review]
Looks good, thank you for your effort, @vladimir-shirmanov 👍

**Review by vbreuss — CHANGES_REQUESTED** [review]
Sorry, I just saw, that the tests failed for Ubuntu and MacOS. Probably the path separator must be OS dependent, @vladimir-shirmanov

**Comment by vladimir-shirmanov** [comment]
> Sorry, I just saw, that the tests failed for Ubuntu and MacOS. Probably the path separator must be OS dependent, @vladimir-shirmanov

I see that some tests are marked as Windows only, how do you think is it a good candidate for that as well? 
Cause drives are Windows-only construct, so it is safe to mark those 2 specific tests as Windows only.

**Review by vbreuss — APPROVED** [review]

**Pull request merged**